### ISSUE TITLE
New Docs: Fatal Error while building docs

### ISSUE THREAD

**Issue by budnix** [handsontable/handsontable#8338]
### Description

When I build documentation(`docs:docker:build`) I've got Fatal Error from Node. It seems that there is a memory leak or the docs generation needs more memory for finishing the task.

This can be reproduced by generating at least three docs versions. In my case I've `next`, `8.4` and `9.0` and executing `docs:docker:build` or `docs:docker:build:production` produces the same error.

```log
#21 151.8 ✔ Server: Compiled successfully in 2.18m
#21 194.3 ✔ Client: Compiled successfully in 2.89m
#21 195.5 wait Rendering static HTML...
#21 207.2
#21 207.2 <--- Last few GCs --->
#21 207.2
#21 207.2 [33:0x6336fc0]   204203 ms: Scavenge 988.1 (1011.3) -> 986.8 (1011.8) MB, 4.2 / 0.0 ms  (average mu = 0.566, current mu = 0.297) allocation failure
#21 207.2 [33:0x6336fc0]   204677 ms: Mark-sweep 988.9 (1012.1) -> 987.0 (1012.6) MB, 467.0 / 0.0 ms  (average mu = 0.400, current mu = 0.112) allocation failure scavenge might not succeed
#21 207.2 [33:0x6336fc0]   205174 ms: Mark-sweep 989.6 (1012.8) -> 987.8 (1017.1) MB, 488.5 / 0.0 ms  (average mu = 0.244, current mu = 0.018) allocation failure scavenge might not succeed
#21 207.2
#21 207.2
#21 207.2 <--- JS stacktrace --->
#21 207.2
#21 207.2 FATAL ERROR: MarkCompactCollector: young object promotion failed Allocation failed - JavaScript heap out of memory
#21 207.2  1: 0xa89e60 node::Abort() [node]
#21 207.2  2: 0x9ade29 node::FatalError(char const*, char const*) [node]
#21 207.2  3: 0xc7583e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
#21 207.2  4: 0xc75bb7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
#21 207.2  5: 0xe3f6d5  [node]
#21 207.2  6: 0xe6ec63 v8::internal::EvacuateNewSpaceVisitor::Visit(v8::internal::HeapObject, int) [node]
#21 207.2  7: 0xe7b816 v8::internal::FullEvacuator::RawEvacuatePage(v8::internal::MemoryChunk*, long*) [node]
#21 207.2  8: 0xe6793f v8::internal::Evacuator::EvacuatePage(v8::internal::MemoryChunk*) [node]
#21 207.2  9: 0xe67bb8 v8::internal::PageEvacuationTask::RunInParallel(v8::internal::ItemParallelJob::Task::Runner) [node]
#21 207.2 10: 0xe599d9 v8::internal::ItemParallelJob::Run() [node]
#21 207.2 11: 0xe7d7a0 void v8::internal::MarkCompactCollectorBase::CreateAndExecuteEvacuationTasks<v8::internal::FullEvacuator, v8::internal::MarkCompactCollector>(v8::internal::MarkCompactCollector*, v8::internal::ItemParallelJob*, v8::internal::MigrationObserver*, long) [node]
#21 207.2 12: 0xe7dfe3 v8::internal::MarkCompactCollector::EvacuatePagesInParallel() [node]
#21 207.2 13: 0xe7e3a5 v8::internal::MarkCompactCollector::Evacuate() [node]
#21 207.2 14: 0xe90b11 v8::internal::MarkCompactCollector::CollectGarbage() [node]
#21 207.2 15: 0xe4c9d8 v8::internal::Heap::MarkCompact() [node]
#21 207.2 16: 0xe4e368 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
#21 207.2 17: 0xe5190c v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
#21 207.2 18: 0xe157da v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
#21 207.2 19: 0x116d5e8 v8::internal::Runtime_AllocateInOldGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
#21 207.2 20: 0x15045f9  [node]
#21 207.5 Aborted
```

### Your environment
* Handsontable version:
* Browser Name and version:
* Operating System:

**Comment by Siemienik** [comment]
I've checked both and it works for me. 
Could you check the docker settings: 
![image](https://user-images.githubusercontent.com/2893099/123921410-436d9700-d987-11eb-89e8-b7e594cf6822.png)

**Comment by budnix** [comment]
In my case, there is a Memory limit set to 2GB. I made some investigation and tests:
 * With 4 versions to generate 8 GB of RAM is ok, decreasing to 7 GB throws a fatal error;
 * 7 GB of RAM is enough to generate no more than 3 versions.

Referring to the [GH Action hardware resources](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) there is 7 GB for Docs deployment action. So we are limited to 3 versions. Moreover, it's worth to mention about this limits in README.

**Comment by Siemienik** [comment]
Ouch, I think we may touch these limits soon. I suppose it needs optimization in the future.

**Comment by Siemienik** [comment]
@kirszenbaum could you update the deployment readme?

**Comment by kirszenbaum** [comment]
@Siemienik - is a brief info on the version limit enough? https://github.com/handsontable/handsontable/pull/8351

**Comment by Siemienik** [comment]
I'm testing this error for many different ways, and I'm trying to find a solution:
https://github.com/handsontable/handsontable/commits/test/how-many-version-can-we-deploy

For the 5 versions released, I do not fount anyone.


Additionally, I run build locally with 15versions and I observe how much memory it needs.
It's increasing memory usage until 3,2GB for **15** versions. It is far away from the 7GB limit.

I'm going to make the same test for the docker build.  This error could be only for docker env related, I guess.

**Comment by Siemienik** [comment]
⚠️ @wojciechczerniak @budnix  ☝️ 
WDYT?

**Comment by Siemienik** [comment]
After research, together with @wojciechczerniak @krzysztofspilka @budnix, we agreed that the best way is to split up the process build for each docs version. Then concatenate build results and put them into a docker image.

**Comment by Siemienik** [comment]
closed as done #8338
